pandas/core/arrays/sparse.py

```diff
@@ -887,21 +887,29 @@ def fillna(self, value=None, method=None, limit=None):
 
         return self._simple_new(new_values, self._sparse_index, new_dtype)
 
-    def shift(self, periods=1):
+    def shift(self, periods=1, fill_value=None):
 
         if not len(self) or periods == 0:
             return self.copy()
 
-        subtype = np.result_type(np.nan, self.dtype.subtype)
+        if fill_value is None:
+            res_type = np.nan
+        else:
+            res_type = fill_value
+
+        subtype = np.result_type(res_type, self.dtype.subtype)
 
         if subtype != self.dtype.subtype:
             # just coerce up front
             arr = self.astype(SparseDtype(subtype, self.fill_value))
         else:
             arr = self
 
+        if fill_value is None:
```

Comment:
why don't you move this clause up to the top, then res_type == fill_value

Reply:
moved